modules/tensor_mechanics/src/materials/RecomputeRadialReturnStressIncrement.C

```diff
@@ -0,0 +1,117 @@
+  Real effective_trial_stress = std::sqrt(3.0 / 2.0 * dev_trial_stress_squared);
+
+  computeStressInitialize(effective_trial_stress);
+
+  // Use Newton sub-iteration to determine the scalar effective inelastic strain increment
+  Real scalar_effective_inelastic_strain = 0;
+  unsigned int iteration = 0;
+  Real residual = 10;
+  Real norm_residual = 10;
+  Real first_norm_residual = 10;
+
+  // create an output string with iteration information when errors occur
+  std::string iteration_output;
+
+  while (iteration < _max_its &&
+        norm_residual > _absolute_tolerance &&
+        (norm_residual/first_norm_residual) > _relative_tolerance)
+  {
+    iterationInitialize(scalar_effective_inelastic_strain);
+
+    residual = computeResidual(effective_trial_stress, scalar_effective_inelastic_strain);
+    norm_residual = std::abs(residual);
+    if (iteration == 0)
+    {
+      first_norm_residual = norm_residual;
+      if (first_norm_residual == 0) // shouldn't be needed here; already set to 10
+        first_norm_residual = 1;
+    }
+
+    Real derivative = computeDerivative(effective_trial_stress, scalar_effective_inelastic_strain);
+
+    scalar_effective_inelastic_strain -= residual / derivative;
+
+    if (_output_iteration_info || _output_iteration_info_on_error)
+    {
+      iteration_output = "In the element " + Moose::stringify(_current_elem->id()) +
+                       + " and the qp point " + Moose::stringify(_qp) + ": \n" +
+                       + " iteration = " + Moose::stringify(iteration ) + "\n" +
+                       + " effective trial stress = " + Moose::stringify(effective_trial_stress) + "\n" +
+                       + " scalar effective inelastic strain = " + Moose::stringify(scalar_effective_inelastic_strain) +"\n" +
+                       + " relative residual = " + Moose::stringify(norm_residual/first_norm_residual) + "\n" +
+                       + " relative tolerance = " + Moose::stringify(_relative_tolerance) + "\n" +
+                       + " absolute residual = " + Moose::stringify(norm_residual) + "\n" +
+                       + " absolute tolerance = " + Moose::stringify(_absolute_tolerance);
+    }
+
+    iterationFinalize(scalar_effective_inelastic_strain);
+    ++iteration;
+  }
+
+  if (_output_iteration_info)
+    _console << iteration_output << std::endl;
+
+  if (iteration == _max_its &&
+     norm_residual > _absolute_tolerance &&
+     (norm_residual/first_norm_residual) > _relative_tolerance)
+  {
+    if (_output_iteration_info_on_error)
+      Moose::err << iteration_output;
+
+    mooseError("Exceeded maximum iterations in RecomputeRadialReturnStressIncrement solve for material: " << _name << ".  Rerun with  'output_iteration_info_on_error = true' for more information.");
+  }
+
+  // compute inelastic strain increments (avoid potential divide by zero - how should this be done)?
```

Comment:
if there is no trial stress what should happen? Should there just be no strain increment? If so then rather than fudging the trial stress I'd just set the _inelastic_strain_increment to zero.